src/librustc_trans/back/write.rs

```diff
@@ -667,7 +667,8 @@ pub fn run_passes(sess: &Session,
 
     // Sanity check
     assert!(trans.modules.len() == sess.opts.cg.codegen_units ||
-            sess.opts.debugging_opts.incremental.is_some());
+            sess.opts.debugging_opts.incremental.is_some() ||
+            output_types.contains_key(&OutputType::Metadata));
```

Comment:
Are you sure that there is no chance that `output_types` may contain not only `Metadata`?